### PR TITLE
Add new diabetes analyses

### PR DESCRIPTION
**Pull request by kurttaylor**
Add new diabetes analyses according to protocol [here on page 5](https://uob.sharepoint.com/:w:/r/teams/grp-ehr/Shared%20Documents/Protocols/post-covid-events/[post-covid-events-diabetes.docx](https://uob.sharepoint.com/:w:/r/teams/grp-ehr/Shared%20Documents/Protocols/post-covid-events/post-covid-events-diabetes.docx?d=w83b1fb0f411f47a68f2724458d541663&csf=1&web=1&e=pbuiRJ)?d=w83b1fb0f411f47a68f2724458d541663&csf=1&web=1&e=pbuiRJ). 

There are quite a few changes, so please ask me if anything doesn't make sense. Please note that I am running everything up to the cox actions (testing new diabetes post hoc analysis with new study definitions) on the real data as we speak [see here.](https://jobs.opensafely.org/university-of-bristol/post-covid-events/diabetes-post-hoc-analysis/logs/) This will give us a good idea if the code is doing what we expect it to. You can access outputs from that job urn in Level 4 if you think it will be useful

### CODE REVIEW

File: analysis/descriptives/diabetes-follow-up-analysis.R [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+input_4$end_date <- as.Date(input_4$end_date)
+
+# Get N with 4 months follow up (those with an end date >= 4 months from t2dm)
+
+input_4 <- input_4 %>% 
+  rowwise() %>%
+  mutate(start_end_diff = as.numeric(difftime(end_date, out_date_t2dm, units = "days"))) %>%
+  ungroup() %>%
+  mutate(start_end_diff_months = start_end_diff/30.417) %>%
+  mutate(follow_4mth = ifelse(start_end_diff_months >= 4, TRUE, FALSE))
```
Comment (marwaalarab) [inline analysis/descriptives/diabetes-follow-up-analysis.R:84]: you could add follow_12mth column here instead of repeating the whole code for 12 mth

**Review by marwaalarab — COMMENTED** [review]
Hi Kurt, 
Great work you've done a lot! really impressive!
I couldn't run the code (see the error on slack) so I wasn't able to test the output. 
No major changes, however, in the new code (diabetes follow up) there is redundancy that could be avoided to make the code more efficient. In the rest the only comment is about using dates directly rather than storing them in variables.

File: analysis/descriptives/diabetes-follow-up-analysis.R [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+  dplyr::select(-c(keep))
+
+# summarise df
+summary(input_4)
+
+# calculate new end date
+
+input_4$cohort_end_date <- cohort_end_date
+input_4$end_date <- apply(input_4[,c("death_date", "dereg_date", "cohort_end_date")],1, min,na.rm=TRUE)
+input_4$end_date <- as.Date(input_4$end_date)
```
[the remark this answers is not in the document]
Comment (kurttaylor) [reply analysis/descriptives/diabetes-follow-up-analysis.R:75]: @marwaalarab I just redid it here so it was explicitly clear in the new study definitions. I think we could also read in the old dates, I will have a think about this. For now, I think it is fine to merge and I can come back to it. Thank you for spotting

File: analysis/descriptives/diabetes-follow-up-analysis.R (hunk 1 above)
Comment (kurttaylor) [reply analysis/descriptives/diabetes-follow-up-analysis.R:84]: Yes I understand what you mean, good point. But I only want to do the 12 month for prevax only `if (cohort_name == "prevax"){` which is why I just did it separately at the bottom. We may not even keep the 12 month analysis in (will see what the collaborators say), so will see what they say first and then go from there.

File: analysis/descriptives/diabetes-follow-up-analysis.R [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+  # (iii) non-hosp COVID-19 infection
+  # (iv) N that were included in the 4-month follow-up analysis and 
+  # (v) N of those that were followed up and still being prescribed medication or had elevated HbA1c.  
+  
+  results <- setNames(data.frame(matrix(ncol = 9, nrow = 0)), c("N_t2dm_any", "N_t2dm_hosp", "N_t2dm_non_hosp",
+                                                                "N_any_COVID_included_12mth", "N_any_COVID_still_treated",
+                                                                "N_hosp_COVID_included_12mth", "N_hosp_COVID_still_treated",
+                                                                "N_non_hosp_COVID_included_12mth", "N_non_hosp_COVID_still_treated"))
+  
+  results[1,1] <- nrow(input_12)
```
[the remark this answers is not in the document]
Comment (kurttaylor) [reply analysis/descriptives/diabetes-follow-up-analysis.R:225]: good idea. Have added your suggestion

**Comment by kurttaylor** [comment]
Hi @marwaalarab, thanks for the comments and taking the time to review, I have made some edits as you will see in my recent commit. I haven't fully addressed everything for now due to time constraints, but if you are happy the code is reflecting the protocol, then please approve. Happy to discuss anything else!

**Review by marwaalarab — APPROVED** [review]

**Pull request merged**